lib/core/src/model.rs

```diff
@@ -951,19 +977,86 @@ pub struct PaymentSwapData {
     pub refund_tx_id: Option<String>,
     pub refund_tx_amount_sat: Option<u64>,
 
+    /// Present only for chain swaps.
+    /// In case of an outgoing chain swap, it's the Bitcoin address which will receive the funds
+    /// In case of an incoming chain swap, it's the Liquid address which will receive the funds
+    pub claim_address: Option<String>,
+
     /// Payment status derived from the swap status
     pub status: PaymentState,
 }
 
+/// The specific details of a payment, depending on its type
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub enum PaymentDetails {
```

Comment:
One benefit is, for some of the binding languages, you can't match directly on the enum and even if you did, you'll need to assign the `details` to a variable to access it's fields. Otherwise it will always be recognized as `PaymentDetails`.